src/script/lua_api/l_object.cpp

```diff
@@ -1720,6 +1720,45 @@ int ObjectRef::l_get_day_night_ratio(lua_State *L)
 	return 1;
 }
 
+// set_time_offset(self, offset)
+int ObjectRef::l_set_time_offset(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	RemotePlayer *player = getplayer(ref);
+	if (!player)
+		return 0;
+
+	int offset_i = luaL_checknumber(L, 2);
+	u16 offset = offset_i;
+	//Negative values
+	if (offset_i < 0) {
+		offset = 24000 + offset_i;
+	}
+
+	player->setTimeOffset(offset);
+	getServer(L)->SendTimeOfDay(player->getPeerId(), 
```

Comment:
IMO this should not be done directly from here
instead of making `SendTimeOfDay` public, add a method in Serve such as `updatePlayerTime()` that does this